### DISCUSSION TITLE
Updating cable terminations via REST API

### DISCUSSION THREAD

**Discussion by jozefws** [netbox-community/netbox#11878] [Q&A] [answered]
Hi all,

Trying to change the terminations on a cable via the API via PATCH request.

I keep getting the same index out of bounds, HTTP 500 error.

Not sure if it a bug or I'm missing something.

Here is the raw HTTP request im sending:
```
curl --location --request PATCH 'https://DOMAIN/api/dcim/cables/4/' \
--header 'Authorization: Token MYTOKEN' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data-raw '{
     "b_terminations": [
        {
          "object_type": "dcim.interface",
          "object_id": 15
        }
    ]
}'
```

And the response is the HTML error with:
```
The complete exception is provided below:

<class 'IndexError'>

list index out of range

Python version: 3.10.6
NetBox version: 3.4.5
```

Any ideas or shall I raise it as an issue?

TIA 😄

**Comment by candlerb** [comment, marked as answer]
A 500 error is a bug,  because if you had provided invalid data, this should have been detected and you should have been given a 4xx response.

It would be helpful if you can get the full backtrace. If you set ADMINS and EMAIL in configuration.py, it should be E-mailed to you.

**Reply by jozefws** [reply]
Okay great thank you - I'll open an issue with the backtrace.

**Reply by jeremystretch** [reply]
Please remember to include _complete_ reproduction steps when submitting a bug (not just the API request).
